Make desktop file valid. See #488912 in Debian BTS.
Index: htop/htop.desktop
===================================================================
--- htop.orig/htop.desktop	2008-11-16 22:08:29.670673088 +0200
+++ htop/htop.desktop	2008-11-16 22:08:32.142641313 +0200
@@ -1,12 +1,10 @@
 [Desktop Entry]
-Encoding=UTF-8
-Version=0.7
+Version=1.0
 Name=Htop
 Type=Application
 Comment=Show System Processes
 Terminal=true
 Exec=htop
-Path=
 Icon=htop
-Categories=ConsoleOnly;System;Application;
+Categories=ConsoleOnly;System;
 GenericName=Process Viewer
