Rename VEID->CTID.
diff -Naur htop-0.8.1.orig/Process.c htop-0.8.1/Process.c
--- htop-0.8.1.orig/Process.c	2008-09-23 05:43:34.000000000 +0200
+++ htop-0.8.1/Process.c	2009-02-24 00:11:35.000000000 +0100
@@ -47,7 +47,7 @@
    PROCESSOR, M_SIZE, M_RESIDENT, M_SHARE, M_TRS, M_DRS, M_LRS, M_DT, ST_UID, PERCENT_CPU, PERCENT_MEM,
    USER, TIME, NLWP, TGID,
    #ifdef HAVE_OPENVZ
-   VEID, VPID,
+   CTID, VPID,
    #endif
    #ifdef HAVE_VSERVER
    VXID,
@@ -124,7 +124,7 @@
    float percent_mem;
    char* user;
    #ifdef HAVE_OPENVZ
-   unsigned int veid;
+   unsigned int ctid;
    unsigned int vpid;
    #endif
    #ifdef HAVE_VSERVER
@@ -163,7 +163,7 @@
    "M_TRS", "M_DRS", "M_LRS", "M_DT", "ST_UID", "PERCENT_CPU", "PERCENT_MEM",
    "USER", "TIME", "NLWP", "TGID", 
 #ifdef HAVE_OPENVZ
-   "VEID", "VPID",
+   "CTID", "VPID",
 #endif
 #ifdef HAVE_VSERVER
    "VXID",
@@ -185,7 +185,7 @@
    " CODE ", " DATA ", " LIB ", " DIRTY ", " UID ", "CPU% ", "MEM% ",
    "USER     ", "  TIME+  ", "NLWP ", " TGID ",
 #ifdef HAVE_OPENVZ
-   " VEID ", " VPID ",
+   " CTID ", " VPID ",
 #endif
 #ifdef HAVE_VSERVER
    " VXID ",
@@ -397,7 +397,7 @@
       break;
    }
    #ifdef HAVE_OPENVZ
-   case VEID: snprintf(buffer, n, "%5u ", this->veid); break;
+   case CTID: snprintf(buffer, n, "%5u ", this->ctid); break;
    case VPID: snprintf(buffer, n, "%5u ", this->vpid); break;
    #endif
    #ifdef HAVE_VSERVER
@@ -566,8 +566,8 @@
    case NLWP:
       return (p1->nlwp - p2->nlwp);
    #ifdef HAVE_OPENVZ
-   case VEID:
-      return (p1->veid - p2->veid);
+   case CTID:
+      return (p1->ctid - p2->ctid);
    case VPID:
       return (p1->vpid - p2->vpid);
    #endif
diff -Naur htop-0.8.1.orig/Process.h htop-0.8.1/Process.h
--- htop-0.8.1.orig/Process.h	2008-09-23 05:43:34.000000000 +0200
+++ htop-0.8.1/Process.h	2009-02-24 00:11:35.000000000 +0100
@@ -49,7 +49,7 @@
    PROCESSOR, M_SIZE, M_RESIDENT, M_SHARE, M_TRS, M_DRS, M_LRS, M_DT, ST_UID, PERCENT_CPU, PERCENT_MEM,
    USER, TIME, NLWP, TGID,
    #ifdef HAVE_OPENVZ
-   VEID, VPID,
+   CTID, VPID,
    #endif
    #ifdef HAVE_VSERVER
    VXID,
@@ -126,7 +126,7 @@
    float percent_mem;
    char* user;
    #ifdef HAVE_OPENVZ
-   unsigned int veid;
+   unsigned int ctid;
    unsigned int vpid;
    #endif
    #ifdef HAVE_VSERVER
diff -Naur htop-0.8.1.orig/ProcessList.c htop-0.8.1/ProcessList.c
--- htop-0.8.1.orig/ProcessList.c	2009-02-24 00:09:37.000000000 +0100
+++ htop-0.8.1/ProcessList.c	2009-02-24 00:11:35.000000000 +0100
@@ -623,7 +623,7 @@
             #ifdef HAVE_OPENVZ
             if (access("/proc/vz", R_OK) != 0) {
                process->vpid = process->pid;
-               process->veid = 0;
+               process->ctid = 0;
             } else {
                snprintf(statusfilename, MAX_NAME, "%s/%s/status", dirname, name);
                status = ProcessList_fopen(this, statusfilename, "r");
@@ -631,17 +631,17 @@
                   goto errorReadingProcess;
                else {
                   char buffer[256];
-                  process->veid = 0;
+                  process->ctid = 0;
                   while (!feof(status)) {
                      char* ok = fgets(buffer, 255, status);
                      if (!ok)
                         break;
 
                      if (String_startsWith(buffer, "envID:")) {
-                        int veid;
-                        int ok = ProcessList_read(this, buffer, "envID:\t%d", &veid);
+                        int ctid;
+                        int ok = ProcessList_read(this, buffer, "envID:\t%d", &ctid);
                         if (ok >= 1) {
-                           process->veid = veid;
+                           process->ctid = ctid;
                         }
                      }
                   }
